src/category_theory/adjunction/fully_faithful.lean

```diff
@@ -63,11 +64,87 @@ instance counit_is_iso_of_R_fully_faithful [full R] [faithful R] : is_iso (adjun
     simp,
   end⟩⟩⟩
 
--- TODO also prove the converses?
--- def L_full_of_unit_is_iso [is_iso (adjunction.unit h)] : full L := sorry
--- def L_faithful_of_unit_is_iso [is_iso (adjunction.unit h)] : faithful L := sorry
--- def R_full_of_counit_is_iso [is_iso (adjunction.counit h)] : full R := sorry
--- def R_faithful_of_counit_is_iso [is_iso (adjunction.counit h)] : faithful R := sorry
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[reducible]
+def whisker_left_L_counit_iso_of_is_iso_unit [is_iso (adjunction.unit h)] : (L ⋙ R ⋙ L) ≅ L :=
+⟨ ⟨(whisker_left L h.counit).app, (whisker_left L h.counit).naturality⟩,
+  ⟨(whisker_right h.unit L).app, (whisker_right h.unit L).naturality⟩,
+  begin
+    ext x, dsimp,
+    rw [comp_hom_eq_id, ←hom_comp_eq_id],
+    simpa [adjunction.right_triangle, functor.id_obj],
+  end, begin
+    ext x, dsimp,
+    simp [adjunction.right_triangle, functor.id_obj],
+  end⟩
```

Comment:
```suggestion
@[simp]
lemma inv_map_unit {X : C} [is_iso (h.unit.app X)] :
  inv (L.map (h.unit.app X)) = h.counit.app (L.obj X) :=
is_iso.inv_eq_of_hom_inv_id (h.left_triangle_components)

/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
by L whiskered with the counit. -/
@[simps]
noncomputable def whisker_left_L_counit_iso_of_is_iso_unit [is_iso (adjunction.unit h)] :
  L ⋙ R ⋙ L ≅ L :=
(functor.associator _ _ _).symm ≪≫
  iso_whisker_right (as_iso (adjunction.unit h)).symm L ≪≫ functor.left_unitor _
```
I think this would be nicer than what you currently have:

- We prefer to have `simps` rather than `reducible` on definitions, essentially this tells Lean to generate simp-lemmas to use the definition rather than expand the definition each time.
- This definition is more compact, but the lemmas generated about it are the same as the ones in your original version, specifically there is a simp lemma `category_theory.whisker_left_L_counit_iso_of_is_iso_unit_hom_app` which says
```
(whisker_left_L_counit_iso_of_is_iso_unit h).hom.app X = h.counit.app (L.obj X)
```
(ordinarily this version would be `inv (L.map (h.unit.app X))` but the simp lemma `inv_map_unit` I added simplifies that further; I think this lemma is actually useful for adjunctions in general and could probably go in adjunction/basic)

Reply:
Thanks! I'll rewrite the other one like this as well.